src/operator/tensor/elemwise_unary_op_basic.cc

```diff
@@ -542,8 +542,7 @@ void SizeComputeCPU(const nnvm::NodeAttrs& attrs,
   CHECK_EQ(req.size(), 1U);
   const TBlob& in_data = inputs[0];
   const TBlob& out_data = outputs[0];
-  const index_t size_var = in_data.Size();
-  memcpy(out_data.dptr_, &size_var, 1U * sizeof(int64_t));
+  *static_cast<int64_t*>(out_data.dptr_) = in_data.Size();
```

Comment:
Hmm...why casting to int64_t in the first place? I think maybe the fix should be just `sizeof(index_t)` instead of `sizeof(int64_t)`? Thanks.

Reply:
If we did that what would happen when the value was read elsewhere in code from dtpr?

Reply:
Let me add some logging code to see what types are used here.  My assumption is that uint64 is used for all size operations.  It's large enough to handle large indexes and it's integer type, which makes sense for floats.  Let me do a few test runs here in CI and validate those assumptions.

Reply:
uint32_t is currently used for index_t in mshadow. I have another PR to change index_t type from uint32_t to int64_t. Please see https://github.com/dmlc/mshadow/pull/348 and https://github.com/apache/incubator-mxnet/pull/11742 for more details

Reply:
Thanks for the background @apeforest.  If it may be changing in the future perhaps it's best to make the size dependent on the TBlob type.  That way we don't have to change this again if index_t changes.